src/local.js

```diff
@@ -0,0 +1,91 @@
+'use strict'
+
+const defaults = require('lodash.defaultsdeep')
+const waterfall = require('async/waterfall')
+const join = require('path').join
+
+const Node = require('./daemon')
+
+const defaultOptions = {
+  config: {
+    'API.HTTPHeaders.Access-Control-Allow-Origin': ['*'],
+    'API.HTTPHeaders.Access-Control-Allow-Methods': [
+      'PUT',
+      'POST',
+      'GET'
+    ],
+    'Addresses.Swarm': [`/ip4/127.0.0.1/tcp/0`],
+    'Addresses.API': `/ip4/127.0.0.1/tcp/0`,
+    'Addresses.Gateway': `/ip4/127.0.0.1/tcp/0`
+  },
```

Comment:
Shouldn't all of these be the default options of IPFS? Do we need to redefine them?

Reply:
This is so that we can start multiple nodes on different random ports, otherwise it uses the default ports and spawning multiple nodes ends up conflicting.

Reply:
So perhaps this should be the default for disposable nodes? It should also be one of the options that gets passed into spawn.